omod/src/test/java/org/openmrs/web/rest/resource/TagResourceTest.java

```diff
@@ -0,0 +1,64 @@
+package org.openmrs.web.rest.resource;
+
+import org.openmrs.Tag;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.api.TagService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResourceTest;
+import org.openmrs.web.rest.TagRestTestConstants;
+
+public class TagResourceTest extends BaseDelegatingResourceTest<TagResource, Tag> {
+	
+	public static final String TAG_RESOURCE_DATASET = "TagResourceDataSet.xml";
+	
+	@Before
+	public void setup() throws Exception {
+		executeDataSet(TAG_RESOURCE_DATASET);
+	}
+	
+	@Override
+	public Tag newObject() {
+		return Context.getService(TagService.class).getTagByUuid(getUuidProperty());
+	}
+	
+	@Override
+	public void validateDefaultRepresentation() throws Exception {
+		super.validateDefaultRepresentation();
+		assertPropEquals("tag", getObject().getTag());
+		assertPropEquals("uuid", getObject().getUuid());
+	}
+	
+	@Override
+	public void validateFullRepresentation() throws Exception {
+		super.validateFullRepresentation();
+		assertPropEquals("tag", getObject().getTag());
+		assertPropEquals("objectType", getObject().getObjectType());
+		assertPropEquals("objectUuid", getObject().getObjectUuid());
+		assertPropEquals("uuid", getObject().getUuid());
+		assertPropPresent("auditInfo");
+	}
+	
+	@Override
+	public void validateRefRepresentation() throws Exception {
+		assertPropEquals("tag", getObject().getTag());
+		assertPropEquals("objectType", getObject().getObjectType());
+		assertPropEquals("objectUuid", getObject().getObjectUuid());
+		assertPropEquals("uuid", getObject().getUuid());
+	}
+	
+	@Override
+	public String getDisplayProperty() {
+		return null;
```

Comment:
this actually refers to adding a display property on the representation, instead of tag. Since we already have a property which displays the name, i didn't add this property. i could replace the tag property with the display property ? @wluyima